component.js

```diff
@@ -95,19 +95,21 @@ module.exports.withDefaults = factory;
 function factory (options) {
   var debug;
   options = options || {};
-  var _shouldComponentUpdate = options.shouldComponentUpdate;
+  var _componentWillMount = options.componentWillMount ||
+                            componentWillMount;
+  var _shouldComponentUpdate = options.shouldComponentUpdate ||
+                               shouldComponentUpdate.withDefaults(options);
+  var _componentWillReceiveProps = options.componentWillReceiveProps ||
```

Comment:
> I was thinking of the order the mixins are called. If we need to have the internal mixins be invoked before any "external" mixins.

I think we wanna run our mixins before others are invoked so they get the same view of `statics` as render and all the followup lifecycle methods would. On the other hand argument can be made that mixins may add stuff to it in order for our mixins to process that too, but that's unlikely I think so I'd rather go ahead with running our mixins first.